backend/modules/process_pool/__init__.py

```diff
@@ -0,0 +1,30 @@
+import asyncio
+from contextlib import asynccontextmanager
+from concurrent.futures import ProcessPoolExecutor
+
+from fastapi import FastAPI
+
+from backend.settings import settings
+
+
+class AsyncProcessPoolExecutor(ProcessPoolExecutor):
+    @staticmethod
+    def _async_to_sync(__fn, *args, **kwarg):
+        loop = asyncio.new_event_loop()
+        result = loop.run_until_complete(__fn(*args, **kwarg))
+        loop.close()
```

Comment:
just `asyncio.run(__fn(*args, **kwargs))` should be good I guess if we only expect to run one coroutine at a time

Reply:
_async_to_sync can be replaced with asyncio.run, yes. I think it does the same thing under the hood, when using it with ProcessPoolExecutor. For threads, it re-uses the existing event loop